File: JAVA/ITES.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.LinkedList;
import java.util.Queue;


public class ITES {
	
	public static class LCG {
		private long seed;
		public LCG (){seed = 1983;}
		public long next() {
			long ret = seed;
			seed = (long) (((seed * 214013) + 2531011) % Math.pow(2, 32));
			return ret % 10000 + 1;
		}
	}
	
	public static void main(String[] args) {
		try {
			BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
			BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
			int cases = Integer.parseInt(reader.readLine());
			
			while (cases-- > 0) {
				String[] input = reader.readLine().split(" ");
				int k = Integer.parseInt(input[0]), n = Integer.parseInt(input[1]);
				writer.append(String.valueOf(signalAnalysis(k, n))); writer.append("\n");
			}
			writer.flush();
			writer.close(); reader.close();
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
	
	public static int signalAnalysis(int k, int n) {
		int ret = 0, sum = 0;
		LCG rand = new LCG();
		Queue<Integer> queue = new LinkedList<Integer>();
		for (int i = 0; i < n; i++) {
			int nextSignal = (int) rand.next();
			sum += nextSignal; queue.add(nextSignal);
			while (sum > k) sum -= queue.poll();
			if (sum == k) ret++;
		}
		return ret;
	}
}

File: CPP/ITES.cpp
#include <cstdio>
#include <queue>
using namespace std;

class LCR {
	private :
		unsigned int seed;
	public :
		LCR() : seed(1983){}
		unsigned int next();
};

unsigned int LCR::next() {
	unsigned int ret = seed;
	seed = seed * 214013u + 2531011u;
	return ret % 10000 + 1;
}

int signalAnalysis(int k, int n) {
	int ret = 0, sum = 0;
	LCR rand;
	queue<int> signals;
	for (int i = 0; i < n; i++) {
		int nextSignal = rand.next();
		sum += nextSignal; signals.push(nextSignal);
		while (sum > k) {
			sum -= signals.front(); signals.pop();
		}
		if (sum == k) ret++;
	}
	return ret;
}

int main() {
	int cases; scanf("%d", &cases);

	while (cases--) {
		int k, n; scanf("%d %d", &k, &n);
		printf("%d\n", signalAnalysis(k, n));
	}
}


